src/test/kotlin/CommonTestCaseProvider.kt

```diff
@@ -12,6 +13,16 @@ class CommonTestCaseProvider : ArgumentsProvider {
         val testsFile = javaClass.classLoader.getResource("jfn-common-test-cases.gen.json")
         val tree = ObjectMapper().readTree(testsFile)
         val testCases = tree.get("testCases") as ArrayNode
+
+        testCases.forEachIndexed { index, it ->
+            val testCase = it as ObjectNode
+            if (tree.has("commonFunctions")) {
+                val commonFunctions = tree.get("commonFunctions") as ArrayNode
+                testCase.set<ArrayNode>("functions", commonFunctions)
+            }
+            testCases.set(index, testCase)
+        }
+
         return testCases.map { Arguments.of(Named.of(it.get("title").textValue(), it)) }.stream()
```

Comment:
We could `read tree.get("commonFunctions")` once , I think this could work? :
```kotlin
        val functions = tree.get("commonFunctions")
        return testCases.map { case ->
            val newCase = if (functions != null) {
                (case as ObjectNode).set<ArrayNode>("functions", functions)
            } else {
                case
            }
            Arguments.of(Named.of(newCase.get("title").textValue(), newCase))
        }.stream()
```

Reply:
Thanks @mtwalli, that's a lot better! Implemented with https://github.com/corona-warn-app/cwa-kotlin-jfn/pull/43/commits/d029686f3fb3ac219bda4b394e163d2ded30863a.